### PR TITLE
[Unitary hack] Improve Docstrings in "task/batch"

### PR DESCRIPTION
**Pull request by shubhusion**
Fixes #957 

Improve function docstrings and Remove markdown errors

### CODE REVIEW

File: src/bloqade/task/batch.py [hunk 1]
```diff
@@ -300,42 +353,18 @@ def get_errors(self, task_indices: Union[List[int], int]):
...
-        "task_errors": [
-            (task_number, task_error)
-            for task_number, task_error in self.task_errors.items()
-        ]
-    }
-
-
-@BatchErrors.set_deserializer
-def _deserialize(obj: dict) -> BatchErrors:
-    return BatchErrors(task_errors=OrderedDict(obj["task_errors"]))
```
Comment (weinbe58) [inline src/bloqade/task/batch.py:327]: why did you delete this code? this is making the CI fail. Please add this code back.

**Review by weinbe58 — CHANGES_REQUESTED** [review]
I appreciate the contribution to the code base but I think it would be a bit beyond the scope of the unitary hack issue. If you want to contribute a change to something other than the docs we should have this be a separate PR.

File: src/bloqade/task/batch.py (hunk 1 above)
Comment (shubhusion) [reply src/bloqade/task/batch.py:327]: I realize I made a mistake while merging conflicts. I was updating the docstrings as part of the unitary hack issue. I will resolve it. Thanks for correcting me.

Comment (weinbe58) [reply src/bloqade/task/batch.py:327]: No problem, if you need help setting up your development environment take a look at our docs page on that: https://bloqade.quera.com/latest/contributing/developing-bloqade/

Comment (weinbe58) [reply src/bloqade/task/batch.py:327]: @shubhusion can I close this PR? I haven't seen any updates on it

**Pull request closed without merging**